internal/credential/state.go

```diff
@@ -196,7 +196,13 @@ func (s *AwsCredentialPersistedState) DeleteCreds(ctx context.Context) error {
 
 // GenerateCredentialChain returns a AWS configuration for the credentials in the state.
 func (s *AwsCredentialPersistedState) GenerateCredentialChain(ctx context.Context) (*aws.Config, error) {
-	return s.CredentialsConfig.GenerateCredentialChain(ctx, s.testOpts...)
+	// Default `WithSharedCredentials` is `true` - This disables the ability to
+	// read credentials from environment variables because we set the default
+	// profile options, so we need to set it to false (see
+	// awsutil@v2.(*CredentialsConfig).generateAwsConfigOptions) for extra
+	// details.
+	opts := append(s.testOpts, awsutil.WithSharedCredentials(false))
```

Comment:
Wont this disable using a shared profile though?

Reply:
I think that this is ok because we technically don't support shared profile credential providers in the plugin yet. Shared profiles can be utilized if a AWS_PROFILE env variable is set by the user in the ec2 instance because of how awsutilv2 works, but I think if we truly want to support shared profiles in the plugin we should have it as an attribute that we can parse and pass down to awsutilv2.

Reply:
> Shared profiles can be utilized if a AWS_PROFILE env variable is set by the user in the ec2 instance

I've just tested this and can confirm it works